google_guest_agent/network/manager/systemd_networkd_linux_test.go

```diff
@@ -0,0 +1,549 @@
+	returnValue bool
+}
+
+// systemdMockRunner is the Mock Runner to use for testing.
+type systemdMockRunner struct {
+	// Options for when running `networkctl --version`
+	versionOpts versionOpts
+
+	// Options for running `systemctl is-active systemd-networkd.service`
+	// isActiveErr indicates whether to return an error when running the command.
+	isActiveErr bool
+
+	// Options for running `networkctl status iface --json=short`
+	statusOpts statusOpts
+}
+
+// versionOpts are options for running `networkctl --version`.
+type versionOpts struct {
+	// returnErr indicates whether the command should return an error.
+	returnErr bool
+
+	// version indicates the version to return when running the command.
+	version int
+}
+
+// statusOpts are options for running `networkctl status iface --json=short`
+type statusOpts struct {
+	// returnValue indicates whether to return a configured or non-configured interface.
+	returnValue bool
+
+	// returnErr indicates whether to return an error.
+	returnErr bool
+
+	// hasKey determines whether the configuredKey should be included or not.
+	hasKey bool
+
+	// configuredKey is used only when returnValue is not err. This indicates what key to
+	// use for determining the configured state.
+	configuredKey string
+}
+
+// runnerOpts are options to set for intializing the MockRunner.
+type runnerOpts struct {
+	// Options for when running `networkctl --version`
+	versionOpts versionOpts
+
+	// Options for running `systemctl is-active systemd-networkd.service`
+	// isActiveErr indicates whether to return an error when running the command.
+	isActiveErr bool
+
+	// Options for running `networkctl status iface --json=short`
+	statusOpts statusOpts
+}
+
+func (s systemdMockRunner) Quiet(ctx context.Context, name string, args ...string) error {
+	// The systemd-networkd implementation only uses Quiet for reloading configurations, so skip
+	// that call.
+	return nil
+}
+
+func (s systemdMockRunner) WithOutput(ctx context.Context, name string, args ...string) *run.Result {
+	if name == "networkctl" && slices.Contains(args, "--version") {
+		verOpts := s.versionOpts
+		if verOpts.returnErr {
+			return &run.Result{
+				ExitCode: 1,
+				StdErr:   "mock error version",
+			}
+		}
+		return &run.Result{
+			StdOut: fmt.Sprintf("systemd %v (%v-1.0)\n+TEST +ESTT +STTE +TTES", verOpts.version, verOpts.version),
+		}
+	}
+	if name == "systemctl" && slices.Contains(args, "is-active") && slices.Contains(args, "systemd-networkd.service") {
+		if s.isActiveErr {
+			return &run.Result{
+				ExitCode: 1,
+			}
+		}
+		return &run.Result{}
+	}
+	if name == "/bin/sh" && slices.Contains(args, "networkctl status iface --json=short") {
+		statusOpts := s.statusOpts
+
+		if statusOpts.returnErr {
+			return &run.Result{
+				ExitCode: 1,
+				StdErr:   "mock error status",
+			}
+		}
+		if statusOpts.returnValue {
+			mockOut := fmt.Sprintf("{\"Name\": \"iface\", \"%s\": \"%s\"}", statusOpts.configuredKey, "configured")
```

Comment:
nit: you can use `%q` instead of `%s` for quotes and avoid escaping `"`

Reply:
I opted for using `\`` quotes instead.